packages/api/models/granules.js

```diff
@@ -375,13 +386,37 @@ class Granule extends Manager {
     return new GranuleSearchQueue(params);
   }
 
+  /**
+   * Delete a granule
+   *
+   * @param {Object} granule record
+   * @returns {Promise}
+   */
+  async delete(granule) {
+    if (granule.published) {
+      throw new DeletePublishedGranule('You cannot delete a granule that is published to CMR. Remove it from CMR first');
+    }
+
+    // Delete granule files
+    await pMap(
```

Comment:
So if a file deletion fails, then the granule deletion will fail, right? And the entry in the files table is deleted in here?

Reply:
Nevermind, I have made my way to the tests!